spec/integration/aws_internet_gateway_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe Chef::Resource::AwsInternetGateway do
+  extend AWSSupport
+
+  when_the_chef_12_server "exists", organization: 'foo', server_scope: :context do
+    with_aws "with a VPC and an internet gateway" do
+      vpc = nil
+      internet_gateway = nil
+
+      before {
+        vpc = driver.ec2.vpcs.create('10.0.0.0/24')
+      }
+
+      it "aws_internet_gateway 'test_internet_gateway' with no parameters" do
+        expect_recipe {
+          aws_internet_gateway 'test_internet_gateway'
+        }.to create_an_aws_internet_gateway('test_internet_gateway').and be_idempotent
+      end
+
+      it "aws_internet_gateway 'test_internet_gateway' with attached vpc" do
+        expect_recipe {
+          aws_internet_gateway 'test_internet_gateway' do
+            vpc vpc.id
+          end
+        }.to create_an_aws_internet_gateway('test_internet_gateway').and be_idempotent
+        filters = [
```

Comment:
Instead of these filters you should be able to specify
```ruby
}.to create_an_aws_internet_gateway('test_internet_gateway', vpc: test_internet_gateway.aws_object).and be_idempotent